### DISCUSSION TITLE
Detect when " Convolution effects cannot be merged " happens

### DISCUSSION THREAD

**Discussion by vis-prime** [pmndrs/postprocessing#492] [Q&A] [answered]
Whats an easy way to detect a convolution effect exists in composer.passes so that I can make programmatically make a new effectPass when adding another convolution effect ?

eg: Chromatic Abberration and SMAA

**Comment by vanruesc** [comment, marked as answer]
Sorry for the late response! I don't think there's a way to detect this currently. You could try copy&pasting the EffectAttribute check into your app as a workaround:

```ts
let attributes: EffectAttribute = EffectAttribute.NONE;

for(const effect of effects) {

	if(attributes & effect.getAttributes() & EffectAttribute.CONVOLUTION) !== 0) {

		// Convolution effect conflict.

	}

	attributes |= effect.getAttributes();

}
```

I'll add a way to access this information in v7.

**Comment by vanruesc** [comment]
> I'll add a way to access this information in v7.

FYI: In v7 it's possible to detect whether effects can be merged by simply attempting to set `effectPass.effects` inside a try-catch block. The setter will throw if the combination is invalid.

https://github.com/pmndrs/postprocessing/blob/7e61527fcb6bb709c669e735ce0ded08c7ceefe5/src/passes/EffectPass.ts#L111-L136
